src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -6122,6 +6114,750 @@ .maxstack  3
                 """);
         }
 
+        [CombinatorialData]
+        [Theory]
+        public void SynthesizedReadOnlyArray([CombinatorialValues("IEnumerable<object>", "IReadOnlyCollection<object>", "IReadOnlyList<object>")] string targetType)
+        {
+            string source = $$"""
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using static System.Console;
+                class Program
+                {
+                    static void Main()
+                    {
+                        Report([1, 2, null]);
+                    }
+                    static void Report({{targetType}} x)
+                    {
+                        Write("IEnumerable.GetEnumerator(): ");
+                        ((IEnumerable)x).Report(includeType: true);
+                        WriteLine();
+                        Write("IEnumerable<object>.GetEnumerator(): ");
+                        ((IEnumerable<object>)x).Report(includeType: true);
+                        WriteLine();
+                        WriteLine("IReadOnlyCollection<object>.Count: {0}", ((IReadOnlyCollection<object>)x).Count);
+                        WriteLine("IReadOnlyList<object>.this[]: {0}", ((IReadOnlyList<object>)x)[1]);
```

Comment:
nit: would be more clear to say `this[1]` here, and updating expected output accordingly, of course.